text/0000-extern-existential-type.md

```diff
@@ -0,0 +1,252 @@
+- Feature Name: extern-existential-type
+- Start Date: 2018-6-29
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+An extension of [#2071](https://github.com/rust-lang/rfcs/pull/2071)'s `existential type` where the definition can live in a different crate than the declaration, rather than the same module.
+This is a crucial tool untangling for untangling dependencies within `std` and other libraries at the root of the ecosystem concerning global resources.
+
+# Motivation
+[motivation]: #motivation
+
+We have a number of situations where one crate defines an interface, and a different crate implements that interface with a canonical singleton:
```

Comment:
Another motivating example https://internals.rust-lang.org/t/global-executors/11295

Reply:
Thanks for letting me know!

Reply:
That was discussed back then here (https://github.com/rust-lang/rfcs/pull/2492#issuecomment-404649166) and here (https://github.com/rust-lang/rfcs/pull/2492#issuecomment-404670971).

Reply:
@gnzlbg Well predicted :)

Reply:
@cramertj is the one that realized that this could be useful for futures